R/importM.COSMIC.R

```diff
@@ -0,0 +1,348 @@
+    hgncSym <- c()
+
+    for (i in 1:nrow(dataCNV)) {
+
+        # Generate all barcodes for the mutations in dataCNV
+        barcodes <- append(barcodes,
+                           paste("TCGA-",
+                                 dataCNV[i,]$ID_SAMPLE, "-",
+                                 dataCNV[i,]$ID_TUMOUR,
+                                 sep = ""))
+
+        # Generate all HGNC symbols using fastMap()
+        hgncSym <- append(hgncSym,
+                          fastMap(dataCNV[i,]$ID_GENE,
+                                  fastMapCosmic,
+                                  type = "Cosmic",
+                                  dev = TRUE))
+
+    }
+
+    # Create a matrix for storing CNV counts.
+    #     nrow is equal to number of unique HGNC symbols
+    #     ncol is equal to the number of unique barcodes.
+    rCNA <- matrix(nrow = length(unique(hgncSym)),
+                   ncol = length(unique(barcodes)))
+
+    # Set the row and column names of rCNA to the
+    # unique HGNC symbols and unique barcodes
+    rownames(rCNA) <- unique(hgncSym)
+    colnames(rCNA) <- unique(barcodes)
+
+    # Populate rCNA matrix with CNA counts
+    for (i in 1:nrow(dataCNV)) {
+        rCNA[fastMap(dataCNV[i,]$ID_GENE,
+                     fastMapCosmic,
+                     type = "Cosmic",
+                     dev = TRUE),
+             paste("TCGA-",
+                   dataCNV[i,]$ID_SAMPLE, "-",
+                   dataCNV[i,]$ID_TUMOUR,
+                   sep = "")] <- as.numeric(dataCNV[i,]$TOTAL_CN)
+    }
+
+    if (!silent) {
+        cat("rCNA object successfully generated.\n")
+    }
+
+
+    # ==== WRITE TO LOG =======================================================
+
+    if(writeLog) {
+
+        myTitle <- "importM.COSMIC"
+
+        # Compile function call record
+        myCall <- character()
+        myCall[1] <- "importM.COSMIC("
+        myCall[2] <- sprintf("fnameCNV = \"%s\", ", fNameCNV)
+        if (writeToFile) {
+            myCall[3] <- sprintf("outFName = \"%s\", ", outFName)
+        } else {
+            myCall[3] <- ""
+        }
+        myCall[4] <- sprintf("writeToFile = %s, ", as.character(writeToFile))
+        myCall[5] <- sprintf("silent = %s, ", as.character(silent))
+        myCall[6] <- sprintf("writeLog = %s)", as.character(writeLog))
+        myCall <- paste0(myCall, collapse = "")
+
+        # Record progress information
+        myNotes <- character()
+        myNotes <- c(myNotes, sprintf("Read %s CNAs from file.", nrow(dataCNV)))
+        if (writeToFile)
+            myNotes <- c(myNotes, sprintf("Wrote rCNA object to %s.", outFName))
+
+        # indicate output object name(s)
+        # myOutput = c("rCNA")
+
+        # send info to log file
+        logEvent(eventTitle = myTitle,
+                 eventCall = myCall,
+                 notes = myNotes
+                 # output = myOutput
+        )
+    }
+
+    # Write to RDS file if specified
+    if (writeToFile) {
```

Comment:
Sorry, a bit confused. I know my module must return an RDS, which is not optional. But I wanted to make writing to file, in addition to returning the RDS object, optional.